assets/js/components/notifications/FirstPartyModeSetupBanner.test.js

```diff
@@ -0,0 +1,248 @@
+import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../googlesitekit/constants';
+import { DEFAULT_NOTIFICATIONS } from '../../googlesitekit/notifications/register-defaults';
+import { CORE_NOTIFICATIONS } from '../../googlesitekit/notifications/datastore/constants';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { withNotificationComponentProps } from '../../googlesitekit/notifications/util/component-props';
+import { enabledFeatures } from '../../features';
+
+const FPM_SETUP_BANNER_NOTIFICATION = 'first-party-mode-setup-cta-banner';
+
+describe( 'FirstPartyModeSetupBanner', () => {
+	let registry;
+
+	const notification = DEFAULT_NOTIFICATIONS[ FPM_SETUP_BANNER_NOTIFICATION ];
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		enabledFeatures.add( 'firstPartyMode' );
+
+		provideSiteInfo( registry );
+		provideUserInfo( registry );
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+			{
+				slug: 'ads',
+				active: true,
+				connected: true,
+			},
+		] );
+
+		registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+			isEnabled: false,
+			isFPMHealthy: true,
+			isScriptAccessEnabled: true,
+		} );
+
+		registry
+			.dispatch( CORE_NOTIFICATIONS )
+			.registerNotification(
+				FPM_SETUP_BANNER_NOTIFICATION,
+				notification
+			);
+
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+	} );
+
+	describe( 'checkRequirements', () => {
+		it( 'is active', async () => {
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( true );
+		} );
+
+		it( 'is not active when FPM is enabled', async () => {
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: true,
+				isFPMHealthy: true,
+				isScriptAccessEnabled: true,
+			} );
+
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( false );
+		} );
+
+		it( 'is not active when FPM is not healthy', async () => {
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: false,
+				isFPMHealthy: false,
+				isScriptAccessEnabled: true,
+			} );
+
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( false );
+		} );
+
+		it( 'is not active when script access is not enabled', async () => {
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: false,
+				isFPMHealthy: true,
+				isScriptAccessEnabled: false,
+			} );
+
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( false );
+		} );
+	} );
+
+	describe( 'Rendering', () => {
```

Comment:
We generally prefer to avoid unnecessary grouping using `describe()` in our tests and keeping a flatter structure.

I'd say the "Rendering" and "onCTAClick" `describe()` blocks are not needed here and can be removed with their `it()` cases flattened in the top `describe()` block.